gapic-generator-cloud/templates/cloud/wrapper_gem/client_test.erb

```diff
@@ -8,22 +8,28 @@ require "gapic/grpc"
 
 class <%= gem.namespace %>::ClientConstructionMinitest < Minitest::Test
 <%- gem.services.each do |service| -%>
-  <%- client_class_name_suffix = service.generate_rest_clients? ? "Rest::#{service.client_name}" : service.client_name -%>
-  def test_<%= service.factory_method_name %>
+<%- if gem.generate_grpc_clients? -%>
+  def test_<%= service.factory_method_name %>_grpc
     Gapic::ServiceStub.stub :new, :stub do
-      <%- if service.generate_rest_clients? -%>
-      client = <%= gem.namespace %>.<%= service.factory_method_name %> do |config|
-        config.credentials = :dummy_credentials
-      end
-      <%- else -%>
       grpc_channel = GRPC::Core::Channel.new "localhost:8888", nil, :this_channel_is_insecure
-      client = <%= gem.namespace %>.<%= service.factory_method_name %> do |config|
+      client = <%= gem.namespace %>.<%= service.factory_method_name %><% if gem.supports_multiple_transports? %> transport: :grpc<% end %> do |config|
         config.credentials = grpc_channel
       end
-      <%- end -%>
-      assert_kind_of <%= gem.namespaced_class "#{service.module_name}::#{client_class_name_suffix}", version: gem.default_version %>, client
+      assert_kind_of <%= gem.namespaced_class "#{service.module_name}::#{service.client_name}", version: gem.default_version %>, client
+    end
+  end
+
+<%- end -%>
+<%- if gem.generate_rest_clients? -%>
+  def test_<%= service.factory_method_name %>_rest
+    Gapic::ServiceStub.stub :new, :stub do
```

Comment:
Strange that we need a gRPC class here

Reply:
Yeah good catch, it should be a different class (`Gapic::Rest::ClientStub`) for rest. Fixed.